src/libraries/System.Diagnostics.Process/src/System/Diagnostics/ProcessWaitState.Unix.cs

```diff
@@ -582,6 +584,21 @@ private bool TryReapChild(bool configureConsole)
                 {
                     // Process is still running
                 }
+                else if (s_allowReapingByOthers && Interop.Sys.GetLastError() == Interop.Error.ECHILD)
```

Comment:
You can avoid duplicating the `waitResult == _processId` handling.

Before that `if` you could add something like:
```cs
if (waitResult == -1 && s_allowReapingByOthers && Interop.Sys.GetLastError() == Interop.Error.ECHILD)
{
    // Some 3rd party library stole our child, pretend we successfully reaped it.
    waitResult = _processId;
    exitCode = xxx;
}
```

Reply:
How about this?

```cs
// Try to get the state of the child process
int exitCode = xxx; //The default value for ambiguous situations.
int waitResult = Interop.Sys.WaitPidExitedNoHang(_processId, out exitCode);

// waitResult == -1 means the child is reaped by other thread(library).
// Set 255 (the highest exit code) and continue execution.
if (waitResult == _processId || waitResult == -1)
{
  _exitCode = exitCode;
```